### PR TITLE
add brief cat matrix doctests

### PR DESCRIPTION
**Pull request by miguelraz**
Brief examples as requested by @vtjnash and Kristoffer on #24873.

### CODE REVIEW

File: base/abstractarray.jl [hunk 1]
```diff
@@ -1829,6 +1829,21 @@ dimensions for every new input array and putting zero blocks elsewhere. For exam
...
+3×3 Matrix{Int64}:
+1  0  0
+0  2  2
+0  2  2
+
+
+julia> cat(dims=[1, 2], [1 2], [3 4])
+2×4 Matrix{Int64}:
+1  2  0  0
+0  0  3  4
```
Comment (mcabbott) [inline base/abstractarray.jl:1845]: Should all examples show the block behaviour?

In another (closed) PR my not-quite-as-compact suggestion was:
```suggestion
julia> cat(ones(2,2), fill(pi,2), zeros(2,3,1); dims=2)
2×6×1 Array{Float64, 3}:
[:, :, 1] =
 1.0  1.0  3.14159  0.0  0.0  0.0
 1.0  1.0  3.14159  0.0  0.0  0.0
 
julia> cat([true], trues(2,2), trues(2,4); dims=(1,2))
5×7 Matrix{Bool}:
 1  0  0  0  0  0  0
 0  1  1  0  0  0  0
 0  1  1  0  0  0  0
 0  0  0  1  1  1  1
 0  0  0  1  1  1  1

File: base/abstractarray.jl [hunk 2]
```diff
@@ -1829,6 +1829,22 @@ dimensions for every new input array and putting zero blocks elsewhere. For exam
 `cat(matrices...; dims=(1,2))` builds a block diagonal matrix, i.e. a block matrix with
 `matrices[1]`, `matrices[2]`, ... as diagonal blocks and matching zero blocks away from the
 diagonal.
+
```
Comment (mcabbott) [inline base/abstractarray.jl:1832]: Maybe it should also link to its friends?
```suggestion

See also [`hcat`](@ref), [`vcat`](@ref), [`hvcat`](@ref), [`repeat`](@ref).

```
(I would include `diagm` except I'm not sure the link will work.)

File: base/abstractarray.jl [hunk 3]
```diff
@@ -1830,23 +1830,22 @@ dimensions for every new input array and putting zero blocks elsewhere. For exam
...
- 13  14
- 15  16
+# Examples
+```jldoctest
+julia> cat([1 2; 3 4], [pi, pi], fill(10, 2,3,1); dims=2)
+2×6×1 Array{Float64, 3}:
+[:, :, 1] =
+ 1.0  2.0  3.14159  10.0  10.0  10.0
+ 3.0  4.0  3.14159  10.0  10.0  10.0
+
```
Comment (simeonschaub) [inline base/abstractarray.jl:1842]:
```suggestion

```

Comment (mcabbott) [reply base/abstractarray.jl]: Sorry about the space, but won't it need a newline here?

Comment (simeonschaub) [reply base/abstractarray.jl]: Oops! :)

File: base/abstractarray.jl [hunk 4]
```diff
@@ -1830,23 +1830,21 @@ dimensions for every new input array and putting zero blocks elsewhere. For exam
...
-[:, :, 4] =
- 13  14
- 15  16
+# Examples
+```jldoctest
+julia> cat([1 2; 3 4], [pi, pi], fill(10, 2,3,1); dims=2)
+2×6×1 Array{Float64, 3}:
+[:, :, 1] =
+ 1.0  2.0  3.14159  10.0  10.0  10.0
+ 3.0  4.0  3.14159  10.0  10.0  10.0
```
Comment (simeonschaub) [inline base/abstractarray.jl:1841]:
```suggestion
 3.0  4.0  3.14159  10.0  10.0  10.0
 
```

File: base/abstractarray.jl (hunk 3 above)
Comment (simeonschaub) [reply base/abstractarray.jl:1842]: Let's hope I didn't screw it up this time...

File: base/abstractarray.jl (hunk 4 above)
Comment (simeonschaub) [reply base/abstractarray.jl:1841]: Thanks GitHub! I definitely wanted a whitespace on that new line again...

**Pull request merged**